js/Learn JavaScript concepts by building a random quote generator.md

```diff
@@ -0,0 +1,289 @@
+<img src="icons/favicon.png">
+</div>
+`;
+```
+*I added images to make it more appealing. 
+*
+### Query Selectors
+To access an element in dom([Document Object Model](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Introduction)), we will use the querySelector() method.
+The Document method querySelector() returns the first element within the document that matches the specified selector. The syntax is 
+```javascript
+element = document.querySelector(selectors);
+```
+ In our HTML file, we have a div element with id="quote". We will access it in the same way as this. And then we will add the HTML template we created earlier to this div element.
+
+### Element.innerHTML
+Each HTML element has an innerHTML property that defines both the HTML code and the text that occurs between that element's opening and closing tag. By changing an element's innerHTML after some user interaction, we can make much more interactive pages.
+
+
+```javascript
+let div=document.querySelector('#quote');
+div.innerHTML=quote;
+```
+The code we wrote till now will work only once. And we want to execute this every time we want to display a quote. We can do this using functions.
+
+### Functions 
+
+> 
+A function is a block of code that performs an action or returns a value. Functions are custom code defined by programmers that are reusable, and can, therefore, make your programs more modular and efficient.
+
+### Defining a Function
+Functions are defined, or declared, with the function keyword. Below is the syntax for a function in JavaScript.
+```javascript
+function nameOfFunction() {
+    // Code to be executed
+}
+```
+We will create the displayQuote function add all this code in the function.
+```javascript
+function displayQuote(){
+let index=Math.floor(Math.random()*quotes.length);
+
+//display the quote of that index
+let div=document.querySelector('#quote');
+let quote=`<div class="card">
+<img src="icons/favicon.png">
+<p>${quotes[index]}</p>
+<img src="icons/favicon.png">
+</div>
+`;
+div.innerHTML=quote;
+}
+```
+We are almost done. Just need to invoke this displayQuote function whenever we click the generate button.
+
+### Events in javaScript
+Events are actions that take place in the browser that can be initiated by either the user or the browser itself. 
+Like in our case it's when the user clicks the generate button.
+
+### Event listeners
+An event listener watches for an event on an element. We will use the addEventListener() method to listen for the event. addEventListener() takes two mandatory parameters . 
+- the event it is listening for 
+- and the listener callback function.
+
+```javascript
+target.addEventListener(type, listener [, options]);
+```
+We have a button element with class="btn". The click event will call the displayQuote function every time we click on the button and display a random quote.
+
+```javascript
+let btn=document.querySelector('.btn');
+btn.addEventListener('click', displayQuote);
+
```

Comment:
what if instead of using event listeners you add an `onclick` method on the button and call this function? This will help keep things simple and in the scope of this post.